Proxy Settings

fictitiouswizard opened the pull request:
Added the ability to specify a proxy server to use when downloading driver files

pep8speaks commented:
Hello @fictitiouswizard! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





* In the file [`webdriver_manager/driver.py`](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py):

> [Line 69:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L69): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 79 characters)
> [Line 73:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L73): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (103 > 79 characters)
> [Line 81:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L81): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 79 characters)
> [Line 83:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L83): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 97:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L97): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 111:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L111): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 133:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L133): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 239:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L239): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 260:42](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L260): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 277:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L277): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (125 > 79 characters)
> [Line 279:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L279): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 288:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/driver.py#L288): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)

* In the file [`webdriver_manager/firefox.py`](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/firefox.py):

> [Line 16:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/firefox.py#L16): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 79 characters)







* In the file [`webdriver_manager/utils.py`](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/utils.py):

> [Line 93:80](https://github.com/SergeyPirogov/webdriver_manager/blob/828f354d03f50898b98039998a66dd5fc5515e73/webdriver_manager/utils.py#L93): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (118 > 79 characters)

aleksandr-kotlyar commented:
It's not only about proxy settings - it's also about browser_version but you didn't wrote about it in this PR and this makes a bunch of questions:
- Was `browser_version` added by mistake?
- If not - why does it need when you can set webdriver version?
- And where is the description in PR about it?

fictitiouswizard commented:
I added browser version because my machine did not have the rights to run the commands to check the registry for the browser version of the driver that needed to be installed so I needed a way to set it manually and skip that bit of code.  It looked like the version variable was used to set the version number of the driver not the version number of the browser so I added it.
